src/util/util.c

```diff
@@ -971,6 +971,36 @@ bool is_user_or_group_name(const char *sudo_user_value)
     return true;
 }
 
+bool is_syntactically_correct_domain_name(const char *domain)
+{
+    size_t i;
+    char c;
+
+    if ((domain == NULL) || (domain[0] == 0)) {
```

Comment:
Why (domain[0] == 0) instead of (domain[0] == '\0')? '\0' sounds more like string related value here.

Reply:
I think 0 is idiomatic enough in the context of C null terminated strings.